### DISCUSSION TITLE
TGIK this Friday on Flux v2?

### DISCUSSION THREAD

**Discussion by evankanderson** [fluxcd/flux2#1108] [Show and tell]
Hi,

I'm planning to explore Flux v2 this week on [TGIK](http://tgik.io/). I'm tracking progress [here](https://github.com/vmware-tanzu/tgik/issues/369); the show will be Friday, March 19 at 1pm-2:30-ish Pacific time (daylight savings in effect).

Anyone interested in providing guidance / suggestions as to particular starting points for learning about Flux?

Happy to take this to slack to discuss if preferred.

**Comment by squaremo** [comment]
Cool! You can certainly reach people via CNCF Slack -- I recommend the #flux-contributors channel. In the meantime, here are some places to start:

The ["Getting Started" guide](https://toolkit.fluxcd.io/get-started/) walks through a simple flux setup, and doesn't assume much prior knowledge.
Going a bit deeper, here is an example of [creating clusters with multi-tenancy and different environments](https://github.com/fluxcd/flux2-multi-tenancy) (dev, prod); and, [a walkthrough of writing a new controller based on flux's API](https://toolkit.fluxcd.io/dev-guides/source-watcher/).

**Reply by hiddeco** [reply]
You may also want to dive into [the monitoring stack](https://toolkit.fluxcd.io/guides/monitoring/), as that gives you something visual to show.

**Comment by stealthybox** [comment]
Flux is super flexible in how you lay out your code and give different people/teams access to things with Git and k8s RBAC.

Flux supports multi-environment and multi-cluster use-cases well, see this answer:
https://github.com/fluxcd/flux2/discussions/766#discussioncomment-312566

We have a YouTube playlist of diverse demos, easy and complex, that may be helpful:
https://www.youtube.com/playlist?list=PL9lTuCFNLaD3fI_g-NXWVxopnJ0adn65d
We'd certainly add the TGIK episode here afterward

In general, it's important to mention that we've separated fetching **Sources** from how you **Reconcile** the paths/pieces of them :)
Flux has a pull-based security model, but you can set up webhook **Receivers** to push events into Flux.
Flux can execute a declarative dependency tree to bootstrap cluster infra, policy, and apps from a single commit and/or from many repos.
It does declarative Helm upgrades through the Helm SDK. (no helm template)
Flux is extensible because it's modular and Kubernetes native.
